gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageImplTest.java

```diff
@@ -141,16 +142,43 @@ public void writeLargeObject_withSmallUploadChunk() throws IOException {
             resourceId,
             /* partitionSize= */ 5 * 1024 * 1024,
             partitionsCount);
-
     assertObjectContent(helperGcs, resourceId, partition, partitionsCount);
-
     assertThat(trackingGcs.requestsTracker.getAllRequestStrings())
         .containsExactlyElementsIn(
             getExpectedRequestsForCreateObject(
                 resourceId, uploadChunkSize, partitionsCount, partition))
         .inOrder();
   }
 
+  @Test
+  public void writeLargeObject_withSmallUploadChunk_testStatistics() throws IOException {
+    StorageResourceId resourceId = new StorageResourceId(TEST_BUCKET, name.getMethodName());
+
+    int uploadChunkSize = 1024 * 1024;
+    GoogleCloudStorage trackingGcs =
+        new GoogleCloudStorageImpl(
+            getOptionsWithUploadChunk(uploadChunkSize),
+            GoogleCloudStorageTestHelper.getCredential());
+
+    int partitionsCount = 1;
+    byte[] partition =
+        writeObject(trackingGcs, resourceId, /* partitionSize= */ 5 * 1024 * 1024, partitionsCount);
+    List<String> expectedRequests =
+        getExpectedRequestsForCreateObject(resourceId, uploadChunkSize, partitionsCount, partition);
+
+    assertThat(
+            trackingGcs
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(1L);
+
+    assertThat(
+            trackingGcs
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(4L);
```

Comment:
Please assert readContents and http stats after read

Reply:
Updated